### PR TITLE
Update docs, clean up bugs, and rename filterpicker

### PR DESCRIPTION
**Pull request by b1lly**

### CODE REVIEW

File: src/core/errors/errors.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 /** @module Errors */
 
 /**
+ * AnswersBaseError is an extension of the base Error object.
+ * This is the object that is used to when reporting to the server.
+ * @extends Error
+ *
  * Error codes fall into one of four categories:
+ * @extends Error
```
Comment (Delerme) [inline src/core/errors/errors.js]: duplicate @extends

File: src/core/errors/errors.js [hunk 2]
```diff
@@ -28,36 +34,61 @@ export class AnswersBaseError extends Error {
...
+ * @extends AnswersBaseError
+ */
 export class AnswersEndpointError extends AnswersBaseError {
   constructor (message, boundary, causedBy) {
     super(300, message, boundary, causedBy);
   }
 }
 
+/**
+ * AnswersCoreError represents errors for precondition failures
```
Comment (Delerme) [inline src/core/errors/errors.js]: maybe just specify precondition failures in the core, not all precondition failures

**Review by Delerme — CHANGES_REQUESTED** [review]
nice

File: src/core/errors/errors.js (hunk 1 above)
Comment (b1lly) [reply src/core/errors/errors.js]: done.

File: src/core/errors/errors.js (hunk 2 above)
Comment (b1lly) [reply src/core/errors/errors.js]: done.

**Review by Delerme — DISMISSED** [review]

**Comment by Delerme** [comment]
Be sure to squash & rebase

File: src/ui/components/search/filtersearchcomponent.js [hunk 3]
```diff
@@ -1,11 +1,18 @@
-/** @module FilterPickerComponent */
+/** @module FilterSearchComponent */
```
Comment (Delerme) [inline src/ui/components/search/filtersearchcomponent.js:1]: can you also rename the handlebars template

**Review by Delerme — CHANGES_REQUESTED** [review]

File: src/ui/components/search/filtersearchcomponent.js (hunk 3 above)
Comment (b1lly) [reply src/ui/components/search/filtersearchcomponent.js:1]: done.

**Review by Delerme — APPROVED** [review]

**Pull request merged**